source/common/http/http1/codec_impl.cc

```diff
@@ -448,27 +479,28 @@ size_t ConnectionImpl::dispatchSlice(const char* slice, size_t len) {
 }
 
 void ConnectionImpl::onHeaderField(const char* data, size_t length) {
+  ENVOY_CONN_LOG(trace, "onHeaderField: data={}", connection_, absl::string_view(data, length));
+  // We previously already finished up the headers, these headers are
+  // now trailers
   if (header_parsing_state_ == HeaderParsingState::Done) {
-    // Ignore trailers.
-    return;
+    processing_trailers_ = true;
   }
-
-  if (header_parsing_state_ == HeaderParsingState::Value) {
+  if (header_parsing_state_ == HeaderParsingState::Value ||
+      trailer_parsing_state_ == HeaderParsingState::Value) {
     completeLastHeader();
   }
 
   current_header_field_.append(data, length);
 }
 
 void ConnectionImpl::onHeaderValue(const char* data, size_t length) {
-  if (header_parsing_state_ == HeaderParsingState::Done) {
-    // Ignore trailers.
-    return;
-  }
-
   const absl::string_view header_value = absl::string_view(data, length);
+  ENVOY_CONN_LOG(trace, "onHeaderValue: data={}", connection_, header_value);
+  if (!current_header_map_) {
+    current_header_map_ = std::make_unique<HeaderMapImpl>();
+  }
 
-  if (strict_header_validation_) {
+  if (strict_header_validation_ && !processing_trailers_) {
```

Comment:
Why not do strict checking for trailers?

Reply:
I was not sure if this check applies only applies to headers and not trailers.